### PR TITLE
[FLINK-12897][python][docs] Improve the Python Table API docs by adding more examples

### PR DESCRIPTION
**Pull request by WeiZhong94**
## What is the purpose of the change

*This pull request add more Python Table API and relevant DataStream/DataSet configuration examples on flink docs and python sphinx docs.*

## Brief change log

  - *Add examples for the obscure API in python sphinx docs.*
  - *Add examples for supported python API in flink docs.*
  - *Fix some typo and style in docs.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

File: docs/dev/table/connect.md [hunk 1]
```diff
@@ -170,6 +181,50 @@ tableEnvironment
 {% endhighlight %}
 </div>
 
+<div data-lang="python" markdown="1">
+{% highlight python %}
+table_environment \
+    .connect(  # declare the external system to connect to
+             Kafka()
```
Comment (HuangXingBo) [inline docs/dev/table/connect.md]: The codestyle doesn't satisfy E126 continuation line over-indented for hanging indent and some similar problem in other code. Is there a better codestyle?

File: docs/dev/table/functions.md [hunk 2]
```diff
@@ -316,7 +316,7 @@ value NOT IN (sub-query)
 </table>
 </div>
 
-<div data-lang="java" markdown="1">
+<div data-lang="Java/Python" markdown="1">
```
Comment (HuangXingBo) [inline docs/dev/table/functions.md]: In line 432 "STRING.similar(STRING)" is previous wrong.Can you change it along the way both in this file and corresponding zh.md file?Thank you.
STRING.similar(STRING) -> STRING1.similar(STRING2)

File: docs/dev/table/streaming/query_configuration.md [hunk 3]
```diff
@@ -135,6 +160,16 @@ val qConfig: StreamQueryConfig = ???
 // set idle state retention time: min = 12 hours, max = 24 hours
 qConfig.withIdleStateRetentionTime(Time.hours(12), Time.hours(24))
 
+{% endhighlight %}
+</div>
+<div data-lang="python" markdown="1">
+{% highlight python %}
+
+q_config = ...
```
Comment (HuangXingBo) [inline docs/dev/table/streaming/query_configuration.md]: Add a comment description that the type of variable q_config is "StreamQueryConfig"

File: flink-python/pyflink/table/table.py [hunk 4]
```diff
@@ -452,6 +452,12 @@ def window(self, window):
             If the :func:`~pyflink.table.GroupWindowedTable.group_by` only references a GroupWindow
             alias, the streamed table will be processed by a single task, i.e., with parallelism 1.
 
+        Example:
+        ::
+
+            >>> tab.window(Tumble.over("10.minutes").on("rowtime").alias("w")).group_by("w") \\
+            ...     .select("a.sum as a, w.start as b, w.end as c, w.rowtime as d")
```
Comment (dianfu) [inline flink-python/pyflink/table/table.py]: What about putting the `group_by` statement in a separate line?

File: flink-python/pyflink/table/table_environment.py [hunk 5]
```diff
@@ -596,6 +659,24 @@ def exec_env(self):
...
     def create(cls, stream_execution_environment, table_config=None):
+        """
+        Creates a :class:`TableEnvironment` for a :class:`StreamExecutionEnvironment`
+
+        Example:
+        ::
+
+            >>> env = StreamExecutionEnvironment.get_execution_environment()
+            >>> table_env = StreamTableEnvironment.create(env)
+            >>> table_config = TableConfig()
```
Comment (dianfu) [inline flink-python/pyflink/table/table_environment.py]: Add some comments between these two kinds of tests?

**Comment by WeiZhong94** [comment]
@dianfu @HuangXingBo Thanks for your reviews! I have addressed the comments in the latest commit.

**Comment by dianfu** [comment]
@WeiZhong94  Thanks a lot for the update. +1 from my side.

**Comment by HuangXingBo** [comment]
@WeiZhong94 LGTM.+1 from my side.

**Pull request closed without merging**